CONTRIBUTING.md

```diff
@@ -1,57 +1,295 @@
+- It's okay to have multiple commits if logical. But must keep squashing to a single commit as much as possible for PR review efficiency.
+- Use git commit conventions when adding messages to commits. [(Read Commit Convention Guide)](https://www.conventionalcommits.org/en/v1.0.0/)
+- Ensure your code complies with formatting/linting configs set by the project.
+- Branches must be tested at least with smoke tests manually.
 - Must include test coverage.
 - Must pass CI/CD pipelines.
-- Reqularly pull updates from develeopment branch `dev` by `--rebase`.
+- Regularly pull updates from the development branch `dev` using `--rebase`.
 - Do not merge branches outside of a PR, use rebase.
 - Do not submit to `main`.
-- Undraft and request for review when ready.
+- Undraft and request a review when ready.
 
-# Formatting
-- Uses google typescript prettier settings for the most part.
-- `/*prettier-ignore*/` is often used to bypass really long debug code lines, this is a tip, not an instruction.
+## Coding Standards
+
+- We use Google TypeScript Prettier settings for most of our formatting.
+- `/*prettier-ignore*/` is often used to bypass really long debug code lines. This is a tip, not an instruction.
 - Do not abuse `/*prettier-ignore*/`.
 
-## Using `/* prettier-ignore */`
+## Commit Messages
+
+Good commit messages help reviewers understand the context and history of changes. Follow these guidelines:
+- Use imperative mood in the subject line (e.g., "Fix bug" not "Fixed bug" or "Fixes bug").
+- Limit the subject line to 50 characters.
+- Wrap the body at 72 characters.
+- Include references to issues or PRs when applicable.
 
-### What is `/* prettier-ignore */`?
+## Testing Requirements
 
-`/* prettier-ignore */` is a directive used in Prettier to prevent the code formatter from formatting a specific piece of code. This can be useful when you have lines of code that should remain unchanged for readability, debugging, or other specific reasons.
+- Include test coverage for all new features and bug fixes.
+- Run all existing tests to ensure nothing is broken by your changes.
+- Manual smoke tests should be performed before submitting PRs.
+- Ensure all changes pass CI/CD pipelines.
 
-### When to Use `/* prettier-ignore */`
+## Setting Up Your Development Environment
 
-There are a few common scenarios where you might want to use `/* prettier-ignore */`:
+### Prerequisites
 
-1. **Long Debug Lines**: When you have long lines of code for debugging purposes that you want to keep on a single line for easier readability.
-2. **Special Formatting**: When you have specific formatting that you want to maintain that Prettier would otherwise change.
-3. **Temporary Code**: When you have temporary code that is meant for testing and you don’t want it to be auto-formatted.
+- Node.js (version 18.16.1)
+- npm (version 9.5.1)
+- Docker (optional, for containerized deployment)
 
-### How to Use `/* prettier-ignore */`
+### Installation
```

Comment:
Since we already covered installation instructions in the Readme.md file, we can link to it from here so we reuse the existing content there.